aeneas/src/os/Linux.v3

```diff
@@ -149,6 +160,54 @@ class LinuxTarget extends Target {
 			rt.recordRuntimeCode(0, 0);
 		}
 
+		// Build list of addresses of imported functions, for replacing references
+		// to their Unimplemented stubs.  This is done here so that all later
+		// encoding of addresses in MachProgram will record references to the stubs.
+
+		// hash set of imported names (as imported)
+		var importNames = HashMap<string, void>.new(Strings.hash, Strings.equal);
+		// Map from addresses of imported functions to their symbol table indexes
+		var importAddrs = HashMap<long, int>.new(int.view, long.==);
+		// Map from addresses of imported functions to locations that refer to them;
+		// locations are (virtual address, offset in DataWriter)
+		var importLocs = HashMap<long, List<(long, int)>>.new(int.view, long.==);
+		// Flag to track whether a symbol section is needed
+		var anySymbols = false;
+		if (haveImports) {
+			for (i < prog.ir.classes.length) {
```

Comment:
I think we can probably avoid another iteration over all the components if we roll this traversal into `MachProgram`--it'll also be more target independent that way.